### ISSUE TITLE
DEV: Review pinned requirements in include files

### ISSUE THREAD

**Issue by stefan6419846** [py-pdf/pypdf#2573]
We should review the following pinned dependencies in our include files for the `requirements.txt` files to ensure that we keep being compatible with recent Python versions:

* Recent `fpdf` versions pass CI as well: https://github.com/py-pdf/pypdf/blob/50bb7f26a3574bb92448c2b64d0d125435f27638/requirements/ci.in#L6
* This seems to be pinned as version 2.18.0 dropped support for Python 3.6? https://github.com/py-pdf/pypdf/blob/50bb7f26a3574bb92448c2b64d0d125435f27638/requirements/dev.in#L4
* Enforces a specific Sphinx version? Recent versions work as well, although probably related to `myst_parser`: https://github.com/py-pdf/pypdf/blob/50bb7f26a3574bb92448c2b64d0d125435f27638/requirements/docs.in#L2
* `myst_parser` broke at some point and generated invalid links (#1569). This still seems to be the case. https://github.com/py-pdf/pypdf/blob/50bb7f26a3574bb92448c2b64d0d125435f27638/requirements/docs.in#L3

**Comment by stefan6419846** [comment]
The `myst_parser` issue seems to be related to https://github.com/py-pdf/pypdf/blob/4bdca1628606706f652f02c260c2e7dd53fb2ff7/docs/conf.py#L44 Changing the setting and fixing the links in

```
docs/dev/documentation.md:56: WARNING: 'myst' cross-reference target not found: 'intro.html#commit-messages' [myst.xref_missing]
docs/user/extract-text.md:41: WARNING: 'myst' cross-reference target not found: '../modules/PageObject.html#pypdf._page.PageObject.extract_text' [myst.xref_missing]
docs/user/merging-pdfs.md:52: WARNING: 'myst' cross-reference target not found: '../modules/PdfWriter.html#pypdf.PdfWriter.append' [myst.xref_missing]
docs/user/robustness.md:32: WARNING: 'myst' cross-reference target not found: '../modules/PdfReader.md' [myst.xref_missing]
docs/user/robustness.md:33: WARNING: 'myst' cross-reference target not found: '../modules/PdfWriter.md' [myst.xref_missing]
```

should resolve it. This additionally requires regenerating the `docs.txt` requirements file afterwards.

**Comment by pubpub-zz** [comment]
@stefan6419846 
what is your proposal approach on this issue ?

**Comment by stefan6419846** [comment]
Check the pinned includes linked above and re-generate the compiled requirements in the first step. Based on this, review which changes are required - at least `myst_parser` requires some changes as outlined above (and probably inside the docs as well due to changing the note syntax).

**Comment by pubpub-zz** [comment]
will you have time to propose PR ?

**Comment by stefan6419846** [comment]
I can have a look at this as soon as I find some more time for thoroughly testing this, although I cannot give any guarantees at the moment and thus it might take some weeks until I indeed manage to do so.

**Comment by stefan6419846** [comment]
This is covered and fixed by #2914.
